### DISCUSSION TITLE
How to call "requestUpdate" for slotted children?

### DISCUSSION THREAD

**Discussion by egloffmark** [lit/lit#3798] [Q&A]
Hi There, I am developing a "LitElement" which offers a `<slot>` to add some childrens. I call a passed `load()` function where the Parent or user of my LitElements can update the childs. I noticed that I can not call simple "this.requestUpdate()" on my LitElement to update the slot, this has no influence. Rather I need to ensure that the Parent / User of my LitElement needs to call that by its own

I am wondering if I am doing this correctly or if there any other ways exists to notify / enforce that the slotted children have to call a render update? 

Thank you in advance

in My LitElement "my-element"
```
	render() {
		return html`
		<div id="container" @scroll="${this.loadChilds.bind(this)}">
			<slot></slot>
		</div>`;
	}

        async loadChilds() {
                [...] do some magic stuff
		if(this.load instanceof Function) {
                       await this.load(); // this callback will enforce to load / change Elements in the slot
                       this.requestUpdate(); // --> This will not update the slotted children to re-render, only my own LitElement get re-rendered
                }
       }
```
In my View (Parent)
```
	@property()
	elements : HTMLElement[] = [];

	render() {
		return html`
		<my-element .load="${this.load.bind(this)}">
			${this.elements}
		</my-element>`; 
	}

	async load() {
		for(let i = start; i<=end; i++) {
                        // create an element
			let element = document.createElement("div");
			....
			this.elements.push(element);
		}
		this.requestUpdate();  // --> This will update the slotted childrens and ask to re-render the Parent. Any better way doing this?
	}
```

**Comment by justinfagnani** [comment]
I'm not following what you're trying to do here 100%. Can you make a reproduction on https://lit.dev/playground?

In general you should not need to call `requestUpdate()` on another component. You should set some state that the component responds to. If you want to respond to children changing, you can use the `slotchange` event: https://developer.mozilla.org/en-US/docs/Web/API/HTMLSlotElement/slotchange_event

**Reply by egloffmark** [reply]
Thank you but I like to do it the other way round. The content of the slot need to be updated based on an event which get handled by the element which contains the slot.  See an [example here at the playground ](https://lit.dev/playground/#project=W3sibmFtZSI6InNpbXBsZS1ncmVldGluZy50cyIsImNvbnRlbnQiOiJpbXBvcnQge2h0bWwsIGNzcywgTGl0RWxlbWVudH0gZnJvbSAnbGl0JztcbmltcG9ydCB7Y3VzdG9tRWxlbWVudCwgcHJvcGVydHl9IGZyb20gJ2xpdC9kZWNvcmF0b3JzLmpzJztcblxuQGN1c3RvbUVsZW1lbnQoJ215LXZpZXcnKVxuZXhwb3J0IGNsYXNzIE15VmlldyBleHRlbmRzIExpdEVsZW1lbnQge1xuICBzdGF0aWMgc3R5bGVzID0gY3NzYC5jYXJkIHsgXG4gICAgICAgY29sb3I6IGJsdWU7IFxuICAgICAgIGhlaWdodDogMTAwcHg7IFxuICAgICAgIHdpZHRoOiAxMDBweDsgXG4gICAgICAgYm9yZGVyLXJhZGl1czogM3B4O1xuICAgICAgIGJvcmRlcjogMXB4IGxpZ2h0Ymx1ZSBzb2xpZDtcbiAgICAgICBkaXNwbGF5OiBibG9jaztcbiAgICAgICBhbGlnbi1pdGVtczogY2VudGVyO1xuICAgICAgIHRleHQtYWxpZ246IGNlbnRlcjtcbiAgICAgICBqdXN0aWZ5LWNvbnRlbnQ6IGNlbnRlcjt9YDtcblxuICBjb250ZW50IDogSFRNTEVsZW1lbnRbXSA9IFtdXG4gIFxuICByZW5kZXIoKSB7XG4gICAgcmV0dXJuIGh0bWxgPG15LWVsZW1lbnQgLmxvYWRDb250ZW50PVwiJHt0aGlzLmxvYWQuYmluZCh0aGlzKX1cIj4ke3RoaXMuY29udGVudH08L215LWVsZW1lbnQ-YDtcbiAgfVxuICBcbiAgbG9hZCgpIHtcbiAgICAgIGNvbnNvbGUubG9nKFwibG9hZCgpIGNhbGxlZC4uLlwiKTtcbiAgICAgIGZvcihsZXQgaT0xOyBpPD0zMDsgaSsrKSB7XG4gICAgICAgICBsZXQgZWxlbWVudCA9IGRvY3VtZW50LmNyZWF0ZUVsZW1lbnQoXCJkaXZcIik7XG4gICAgICAgICBlbGVtZW50LmNsYXNzTmFtZSA9IFwiY2FyZFwiO1xuICAgICAgICAgZWxlbWVudC5pbm5lckhUTUwgPSAgXCJjYXJkLVwiICsgaTtcbiAgICAgICAgIHRoaXMuY29udGVudC5wdXNoKGVsZW1lbnQpO1xuICAgICAgfVxuICAgICB0aGlzLnJlcXVlc3RVcGRhdGUoKTtcdC8vIDwtLSB1bmNvbW1lbnQgdG8gbWFrZSBjb250ZW50IHZpc2libGUgd2hlbiBteS1lbGVtZW50XG4gIH1cbn1cblxuQGN1c3RvbUVsZW1lbnQoJ215LWVsZW1lbnQnKVxuZXhwb3J0IGNsYXNzIE15RWxlbWVudCBleHRlbmRzIExpdEVsZW1lbnQge1xuICBzdGF0aWMgc3R5bGVzID0gY3NzYFxuICAgICAgZGl2ICB7XG4gICAgICAgZGlzcGxheTogZmxleDtcbiAgICAgICBmbGV4LXdyYXA6IHdyYXA7XG4gICAgICAgb3ZlcmZsb3cteTogc2Nyb2xsO1xuICAgICAgfVxuICAgICAgZGl2IHtjb2xvcjogcmVkOyBib3JkZXI6IDFweCByZWQgc29saWR9YFxuXG4gIGxvYWRDb250ZW50IDogRnVuY3Rpb247XG4gICAgICBcbiAgcmVuZGVyKCkge1xuICAgIHJldHVybiBodG1sYDxkaXY-PHNsb3Q-PHNsb3Q-PC9kaXY-XG4gICAgICAgIDxicj5cbiAgICAgICAgPGJ1dHRvbiBAY2xpY2s9XCIke3RoaXMuaGFuZGxlQ2xpY2suYmluZCh0aGlzKX1cIj5DbGljayBNZSB0byBsb2FkIENvbnRlbnQgb2YgdGhlIFNsb3Q8L2J1dHRvbj5gO1xuICB9XG5cbiAgaGFuZGxlQ2xpY2soZTogRXZlbnQpIHtcbiAgICAgY29uc29sZS5sb2coXCJoYW5kbGVDbGljaygpIGNhbGxlZC4uLi5cIik7XG4gICAgIGlmKHRoaXMubG9hZENvbnRlbnQgaW5zdGFuY2VvZiBGdW5jdGlvbikge1xuICAgICAgIHRoaXMubG9hZENvbnRlbnQoKTtcbiAgICAgfVxuICB9XG5cbn0ifSx7Im5hbWUiOiJpbmRleC5odG1sIiwiY29udGVudCI6IjwhRE9DVFlQRSBodG1sPlxuPGhlYWQ-XG4gIDxzY3JpcHQgdHlwZT1cIm1vZHVsZVwiIHNyYz1cIi4vc2ltcGxlLWdyZWV0aW5nLmpzXCI-PC9zY3JpcHQ-XG48L2hlYWQ-XG48Ym9keT5cbiAgICA8bXktdmlldz5cbjwvYm9keT5cbiJ9LHsibmFtZSI6InBhY2thZ2UuanNvbiIsImNvbnRlbnQiOiJ7XG4gIFwiZGVwZW5kZW5jaWVzXCI6IHtcbiAgICBcImxpdFwiOiBcIl4yLjAuMFwiLFxuICAgIFwiQGxpdC9yZWFjdGl2ZS1lbGVtZW50XCI6IFwiXjEuMC4wXCIsXG4gICAgXCJsaXQtZWxlbWVudFwiOiBcIl4zLjAuMFwiLFxuICAgIFwibGl0LWh0bWxcIjogXCJeMi4wLjBcIlxuICB9XG59IiwiaGlkZGVuIjp0cnVlfV0)

**Reply by vdegenne** [reply]
@egloffmark Maybe you can use an observer https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver
